lib/matplotlib/text.py

```diff
@@ -1162,10 +1162,12 @@ def set_text(self, s):
 
         It may contain newlines (``\\n``) or math in LaTeX syntax.
 
-        ACCEPTS: string or anything printable with '%s' conversion.
+        ACCEPTS: string or anything printable with '%s' conversion, except
+        ``None``, which leaves the text string empty.
         """
-        self._text = '%s' % (s,)
-        self.stale = True
+        if s is not None:
```

Comment:
shouldn't you always clear the text when s is None?

Reply:
Oh, good question!  Open to suggestions

Reply:
Set to empty string if `None` is passed.

Reply:
There's a lot to say here:

- IMO `'%s' % (s,)` is an anti-pattern. It is identical to `str(s)`.
- `self._text` is not defined in `__init__`. It only calls `set_text(text)`. Something like `Text(text=None).get_text()` would fail with an attibute error in your code.  It's good practice to declare your attributes in __init__. You should add `self._text = ''` to __init__.
- I would expect that `set_text(None)` and `set_text('')` are completely equivalent.
- Minor thing: Add a colon at the end of the sentence in the docstring.

So either always stale:
~~~
def set_text(s):
    self._text = str(s) if s is not None else ''
    self.stale = True
~~~

or stale only when there's really a change:
~~~
def set_text(s):
    if s is None:
        s = ''
    if s != self._text:
        self._text = s
        self.stale = True
~~~

Reply:
Great catch.  You're right I didn't notice that `self._text` wasn't being set.

Reply:
@timhoffm not sure where you want the colon...

Reply:
Sorry, full stop: "Set the text string *s*."

Reply:
Great.  Fixed.  Also, your logic was cleaner, so used that (second option).